services/blockstorage/sync/state_idle_test.go

```diff
@@ -0,0 +1,45 @@
+package sync
+
+import (
+	"github.com/stretchr/testify/require"
+	"testing"
+)
+
+func TestIdleStateStaysIdleOnCommit(t *testing.T) {
+	h := newBlockSyncHarness()
+	idle := h.sf.CreateIdleState()
+	var next syncState = nil
+	latch := make(chan struct{})
```

Comment:
I wonder; this latch-goroutine-processState mechanism repeats in some tests - is there a common utility function that could be extracted or would that be premature?